test-smoke/colony-storage-consistent.js

```diff
@@ -0,0 +1,163 @@
+        if (err !== null) return reject(err);
+        return resolve(res.toString("hex"));
+      });
+    });
+  }
+
+  describe("Smoke tests to check our storage layout does not change", () => {
+    // There are many things you could do that one would expect to change these hashes. If you've made changes that change the contents of storage of a
+    // contract, then these hashes will change. This could include adding an extra transaction somewhere, which could cause the address a contract is
+    // deployed to change, which if it is stored somewhere would cause the state hash of the storage to change.
+
+    // If you haven't touched the contracts, however, and this test fails, then something is afoot.
+    // In theory, the storage slots used by solidity could change in an upgrade,
+    // which this test was written with in mind to try and detect. They don't guarantee this hasn't happened if they pass, but if they fail without just
+    // cause then we need to think very carefully about what's going on.
+
+    it("storage contents should be as expected", async () => {
+      await colony.makeExpenditure(1, UINT256_MAX, 1, { from: ADMIN });
+      const expenditureId = await colony.getExpenditureCount();
+
+      await colony.setExpenditureRecipient(expenditureId, SLOT0, RECIPIENT, { from: ADMIN });
+      await colony.setExpenditurePayout(expenditureId, SLOT0, token.address, WAD, { from: ADMIN });
+      await colony.setExpenditureSkill(expenditureId, SLOT0, GLOBAL_SKILL_ID, { from: ADMIN });
+
+      const expenditure = await colony.getExpenditure(expenditureId);
+      await colony.moveFundsBetweenPots(1, UINT256_MAX, UINT256_MAX, domain1.fundingPotId, expenditure.fundingPotId, WAD, token.address);
+      await colony.finalizeExpenditure(expenditureId, { from: ADMIN });
+      await colony.claimExpenditurePayout(expenditureId, SLOT0, token.address);
+
+      const miningCycleAddress = await colonyNetwork.getReputationMiningCycle(false);
+      const miningCycleStateHash = await getAddressStateHash(miningCycleAddress);
+      const miningCycleAccount = new Account(miningCycleStateHash);
+
+      // For this test to be reproducible, have to zero timestamps / time depenedent things
+      // For colonyNetwork, that means the mining staking timestamp
+
+      const contractEditing = await ContractEditing.new();
+      const networkAsER = await EtherRouter.at(colonyNetwork.address);
+      const colonyNetworkResolverAddress = await networkAsER.resolver();
+      const colonyNetworkResolver = await Resolver.at(colonyNetworkResolverAddress);
+      await colonyNetworkResolver.register("setStorageSlot(uint256,bytes32)", contractEditing.address);
+      const editableNetwork = await ContractEditing.at(colonyNetwork.address);
+
+      let hashable = `0x000000000000000000000000${accounts[5].slice(2)}${new BN(33).toString(16, 64)}`;
+      let hashed = web3.utils.sha3(hashable);
```

Comment:
Not `soliditySha3`?

Reply:
Originally didn't seem to work with `soliditySha3` due to (I think) the way I was expressing the variables, as that function close-packs them before hashing. I've since changed to a big string, however, so I can switch back to `soliditySha3` without issue.